### PR TITLE
Fix the status method in kubernetes provider

### PR DESCRIPTION
**Pull request by ZhuozhaoLi**
This PR modifies the `_status` method in kubernetes provider.

It also changes the site test `test_site.py` a bit because in Kubernetes provider, there cant be more than nodes on one block, so instead there are two blocks, each with one node. The second block takes some time to launch, causing the later assertion `assert len(pinfo) == 2, "Expected two nodes, instead got {}".format(pinfo)` in the test to fail.

### CODE REVIEW

**Comment by yadudoc** [comment]
@BenGalewsky You might want to take a quick look at this. This impacts funcX.

File: parsl/providers/kubernetes/kube.py [hunk 1]
```diff
@@ -195,10 +202,24 @@ def _status(self):
...
+        job_ids = list(self.resources.keys())
+        logger.debug("Polling Kubernetes pod status: {}".format(job_ids))
+        for jid in job_ids:
+            phase = None
+            try:
+                pod_status = self.kube_client.read_namespaced_pod_status(name=jid, namespace=self.namespace)
+            except Exception:
+                logger.warning("Failed to poll pod {} status, most likely because pod was terminated".format(jid))
+                if self.resources[jid]['status'] is JobStatus(JobState.RUNNING):
+                    phase = 'Unknown'
```
[the remark this answers is not in the document]
Comment (ZhuozhaoLi) [reply parsl/providers/kubernetes/kube.py:217]: Should be fine to review now @yadudoc . Kube provider only polls those pods with non-terminal states.

**Review by yadudoc — APPROVED** [review]
ZZ, this PR looks good to go. It's unclear whether the changes to `test_site.py` necessary for this PR. Can you clarify why this was added?

**Comment by ZhuozhaoLi** [comment]
> ZZ, this PR looks good to go. It's unclear whether the changes to `test_site.py` necessary for this PR. Can you clarify why this was added?

The test config for general providers is a block with 2 nodes, while the config for k8s has 2 blocks and each has only 1 node (one cannot request 2 nodes in 1 block for k8s). This is to ensure there will be two workers before real testing (adding a small sleep to ensure even if the `scale_out` is too slow).

**Comment by yadudoc** [comment]
Okay, that makes sense.

**Pull request merged**